openmdao/core/component.py

```diff
@@ -1259,6 +1259,10 @@ def _declare_partials(self, of, wrt, dct):
                 meta['cols'] = cols
                 meta['shape'] = shape = (abs2meta[abs_key[0]]['size'], abs2meta[abs_key[1]]['size'])
 
+                if shape[0] == 0 or shape[1] == 0:
+                    msg = "{}: '{}' is an array of size 0"
+                    raise ValueError(msg.format(self.msginfo, abs_key[0]))
+
```

Comment:
This error message only prints out the variable name `abs_key[0]`, even if `abs_key[1]` is the one that is zero-shape.  You should fix the 2nd test so that it fails, and then fix this bug.

Reply:
Good catch, fixed